### DISCUSSION TITLE
How to change default new file permissions?

### DISCUSSION THREAD

**Discussion by VirgilChen97** [drakkan/sftpgo#1361] [Q&A] [answered]
I noticed that files uploaded via webdav always has permission of 0644. How can I can this to 0660?

**Comment by rel33** [comment]
You may add UMask=0007 parameter to systemd configuration file, and sftpgo would change default umask to 0007 for all operations.
E.g. in ubuntu, config file is in /etc/systemd/system/multi-user.target.wants/sftpgo.service under [Service] you add line (case sensitive)
UMask=0007

Then reload config and restart sftpgo:
 systemctl daemon-reload
 systemctl restart sftpgo

Newly created files would have mode -rw-rw---- (660)

EDITED 2023-08-08: 
There will be additional configuration parameter in sftpgo.json - today it has been added to the source code:
"  - `umask`, string. Set the file mode creation mask, for example `002`. Leave blank to use the system umask. Supported on *NIX platforms. Default: blank."
Thank you @drakkan

**Reply by VirgilChen97** [reply]
Thanks for the solution! But how do I change umask in the offical docker image? I can't find any related documentation, seems like the only way is to build a new image based on the offical one?

**Reply by rel33** [reply]
Hi, no idea. I run sftpgo directly on GNU/Linux installation, not via docker image.

**Comment by VirgilChen97** [comment, marked as answer]
Inspired by @rel33, I built a new image based on the offical one to solve the problem:
- Dockerfile
```dockerfile
FROM drakkan/sftpgo:v2.5.4-alpine
COPY ./entrypoint.sh /
ENTRYPOINT ["/entrypoint.sh"]
```
- entrypoint.sh
```bash
#!/bin/sh
umask 0007
sftpgo serve
```
